### PR TITLE
xh: Submission of new port (v0.9.2)

### PR DESCRIPTION
**Pull request by judaew**
###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 11.2.3 20D91
Xcode 12.4 12D4e

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: net/xh/Portfile [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+    xinstall -d -m 755 ${destroot}${prefix}/share/fish/completions
+    xinstall -m 644 ${worksrcpath}/completions/xh.fish ${destroot}${prefix}/share/fish/completions/
+    xinstall -d -m 755 ${destroot}${prefix}/share/zsh/site-functions
+    xinstall -m 644 ${worksrcpath}/completions/_xh ${destroot}${prefix}/share/zsh/site-functions/
+}
+
+post-destroot {
+    # xh will default to HTTPS scheme if the binary name is one of xhs, https,
+    # or xhttps
+    ln ${destroot}${prefix}/bin/${name} ${destroot}${prefix}/bin/${name}s
```
Comment (herbygillot) [inline net/xh/Portfile:42]: I would say prefer a soft link here, not a hard link (`ln -s`)

File: net/xh/Portfile [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+platforms           darwin
+license             MIT
+maintainers         {outlook.de:judaew @judaew} openmaintainer
+
+description         xh is a friendly and fast tool for sending HTTP requests.
+long_description    ${description} It reimplements as much as possible of \
+                    HTTPie's excellent design.
+homepage            https://github.com/ducaale/xh
+
+checksums           xh-0.9.2.tar.gz \
```
Comment (herbygillot) [inline net/xh/Portfile:20]: Don't repeat yourself... in this case you're repeating the version.

If `checksums ${distname}${extract.suffix}` does not work here, at least use `xh-${version}${extract.suffix}` instead.

File: net/xh/Portfile [hunk 3]
```diff
@@ -0,0 +1,321 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+PortGroup           cargo  1.0
+
+github.setup        ducaale xh 0.9.2 v
+revision            0
+
+categories          net
```
Comment (herbygillot) [inline net/xh/Portfile:10]: Since this is an HTTP tool, maybe add the `www` category (`net www`)

**Comment by judaew** [comment]
Thank so much, done.

**Comment by herbygillot** [comment]
Thanks @judaew

**Comment by judaew** [comment]
I probably make common mistakes, thanks for your patience.

**Pull request merged**